plugins/plugin-chart-table/src/controlPanel.tsx

```diff
@@ -103,12 +112,20 @@ const percent_metrics: typeof sharedControls.metrics = {
   ),
   multi: true,
   visibility: isAggMode,
-  mapStateToProps: ({ datasource, controls }) => ({
+  mapStateToProps: ({ datasource, controls }, controlState) => ({
     columns: datasource?.columns || [],
     savedMetrics: datasource?.metrics || [],
     datasourceType: datasource?.type,
     queryMode: getQueryMode(controls),
+    externalValidationErrors:
+      isAggMode({ controls }) &&
+      ensureIsArray(controls.groupby?.value).length === 0 &&
+      ensureIsArray(controls.metrics?.value).length === 0 &&
+      ensureIsArray(controlState.value).length === 0
+        ? [t('Group By, Metrics or Percentage Metrics must have a value')]
+        : [],
```

Comment:
I'm temped to ask if we could break this out so we don't have to repeat the same logic three times?

Reply:
Done